clients/go/cmd/zbctl/main_test.go

```diff
@@ -155,16 +159,46 @@ func (s *integrationTestSuite) TestCommonCommands() {
 			if err != nil {
 				t.Fatal(err)
 			}
-			want := strings.Split(string(goldenOut), "\n")
-			got := strings.Split(string(cmdOut), "\n")
 
-			if diff := cmp.Diff(want, got, cmp.Comparer(composeComparer(cmpIgnoreNums, cmpIgnoreVersion))); diff != "" {
-				t.Fatalf("%s: diff (-want +got):\n%s", test.name, diff)
+			if json.Valid(goldenOut) {
```

Comment:
Hm, how about having the test specify that it is outputting JSON? So we define all test cases in the table, we could just add a `outputsJson` or something property in there, instead of checking the output? What do you think?

Reply:
I thought about that but I went with this because: a) that's more parameterization to put in the table tests, another thing the developer has to be aware of and configure b) the output should match the golden file so if there's a mismatch the test would already fail
Why do you prefer the additional parameter? I just want to understand the advantage of that approach

Reply:
I favour the extra param because it's explicit and low overhead - when adding a new test, you would look at the other tests, and just fill out the params - I think the overhead or "worry" there is negligible. At the same time, it's explicit - this test expects JSON and will test for it, as opposed to detecting it based on what's in the file, which is sort of magic - very low level magic, granted :sweat_smile:, but still.